### ISSUE TITLE
Volume up and down on STR-DA1000ES

### ISSUE THREAD

**Issue by lnxsrt** [Ircama/Sony_SLink#2]
Thank you for this great code! It has worked great for interfacing with my STR-DA1000ES (3V TTL; 112 AMP device ID).

The volume up and down codes (20 and 21) seem to make the volume level display on the receiver, but doesn't actually change the volume. I am thinking that the initial volume code doesn't really do anything, and it needs to be repeated. I tried this...

`slink.sendCommand(112, 20);`
`slink.sendCommand(112, 20);`
`slink.sendCommand(112, 20);`
`slink.sendCommand(112, 20);`
`slink.sendCommand(112, 20);`

But, that doesn't seem to be working. Any ideas?

**Comment by Ircama** [comment]
Unfortunately I am not aware of STR-DA1000ES S-LINK/Control-A1 codes and I have no related docs.
Volume can anyway be changed with STR-DA50ES.
You might have a look here: http://www.hifi-remote.com/sony/Sony_dsp_192.htm
Did you test it with 192 device ID instead of 112?

**Comment by lnxsrt** [comment]
Yes, I tried 192 and it does not respond at all to that code. On the link you sent, the author mentions that newer receivers use 112 for AMP functions, this seems to be the case with my receiver as well. The command codes line up quite well and most work. 

So I think I figured out the issue with some experimentation. I need to put some amount of wait time between `slink.sendCommand(112, 20);`. Seems it needs to be between 3ms and 1960ms for it to work. Any shorter or longer and the volume won't change.

`void loop() {`
 `  slink.sendCommand(112, 20);`
`  delay(3); //up to delay(1960);`
`}`
